micro-rdk/src/esp32/pin.rs

```diff
@@ -185,34 +185,43 @@ impl Esp32GPIOPin {
         self.interrupt_type.is_some()
     }
 
-    pub fn setup_interrupt(&mut self, intr_type: InterruptType) -> Result<(), BoardError> {
-        match &self.interrupt_type {
-            Some(existing_type) => {
-                if *existing_type == intr_type {
-                    return Ok(());
-                }
-            }
-            None => {
-                self.interrupt_type = Some(intr_type);
-            }
-        };
+    pub(crate) fn setup_interrupt(
+        &mut self,
+        intr_type: InterruptType,
+        cb: gpio_isr_t,
+        arg: *mut core::ffi::c_void,
+    ) -> Result<(), BoardError> {
+        if cb.is_none() {
+            return Err(BoardError::GpioPinOtherError(
+                self.pin as u32,
+                "interrupt callback cannot be `None`".to_string().into(),
+            ));
+        }
+
+        if self.pin == 36 || self.pin == 39 {
```

Comment:
are there masks for determining this exposed by the `wifi` or `adc` headers like we do to determine valid output pins?

Reply:
not that I can find. Espressif doesn't do a hard check for this and only advises not to use them [here](https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-reference/peripherals/gpio.html#_CPPv416gpio_intr_enable10gpio_num_t) and in the doc linked